singer/src/main/java/com/pinterest/singer/processor/MemoryEfficientLogStreamProcessor.java

```diff
@@ -96,7 +99,18 @@ protected int processLogMessageBatch() throws IOException, LogStreamWriterExcept
         // because there is some data to read we need to prepare the commit
         writer.startCommit();
       }
-      writer.writeLogMessageToCommit(message.getLogMessage());
+      LogMessage logMessage = message.getLogMessage();
+      OpenTsdbMetricConverter.gauge(
+              SingerMetrics.PROCESSOR_MESSAGE_KEY_SIZE_BYTES,
+              logMessage.getKey().length,
```

Comment:
Potential NPE when key is null

Reply:
We should also add a test to processor to add null keys to capture and validate the null tests.

Reply:
Do we want unit test for the metrics piece? Or general unit tests to cover null keys?

Reply:
General processor unit test to cover null keys